main.go

```diff
@@ -82,6 +108,155 @@ func runMain() error {
 		fmt.Println("The local master branch is not in sync with the remote master branch. Must be in sync to conduct release process.")
 		return nil
 	}
+	
+	// Guess the next release version
+	latestReleaseVersion := getLatestReleaseVersion(repository, NO_PREVIOUS_VERSION)
+
+	// Conduct changelog file validation
+	changelogFilepath := path.Join(currentWorkingDirectory, relChangelogFilepath)
+
+	tbdHeaderCount := grepFile(changelogFilepath, TBD_VERSION_HEADER_REGEX)
+	if tbdHeaderCount != EXPECTED_NUM_TBD_HEADER_LINES {
+		fmt.Printf("There should be %d TBD header lines in the changelog. Instead there are %d.\n", EXPECTED_NUM_TBD_HEADER_LINES, tbdHeaderCount)
+		return nil
+	}
+
+	versionHeaderCount := grepFile(changelogFilepath, VERSION_HEADER_REGEX)
+	if versionHeaderCount == 0 {
+		fmt.Println("No previous changelog versions were detected in this changelog. Are you sure that the changelog is in sync with the release tags on this branch?")
+		return nil
+	}
+
+	existsBreakingChanges := detectBreakingChanges(changelogFilepath)
+	var nextReleaseVersion semver.Version
+	if existsBreakingChanges {
+		nextReleaseVersion = latestReleaseVersion.IncMinor()
+	} else {
+		nextReleaseVersion = latestReleaseVersion.IncPatch()
+	}
 
+	go wait()
+	fmt.Printf("VERIFICATION: Release new version '%s'? (ENTER to continue, Ctrl-C to quit))", nextReleaseVersion.String())
+	fmt.Scanln()
 	return nil
 }
+
+func wait() {
+    i := 0
+    for {
+        time.Sleep(time.Second * 1)
+        i++
+    }
+}
+
+// adapted from: https://stackoverflow.com/questions/26709971/could-this-be-more-efficient-in-go
+func grepFile(file string, regexPat string) int64 {
+	r, err := regexp.Compile(regexPat)
+	if err != nil {
+		logrus.Errorf("Could not parse regexp: '%s'", regexPat, err)
+	}
+    patCount := int64(0)
+    f, err := os.Open(file)
+    if err != nil {
+		logrus.Errorf("An error occurred while attempting to open file", err)
+    }
+    defer f.Close()
+    scanner := bufio.NewScanner(f)
+    for scanner.Scan() {
+		if r.Match(scanner.Bytes()) {
+			patCount++
+		}
+    }
+    if err := scanner.Err(); err != nil {
+        logrus.Errorf("An error occurred while scanning file.\n%v", err)
+    }
+    return patCount
+}
+
+func getLatestReleaseVersion(repo *git.Repository, noPrevVersion string) semver.Version {
+	semverRegex, err := regexp.Compile(SEMVER_REGEX)
```

Comment:
Tip: compiling a regex is usually a fairly slow operation so people will usually do it on instantiation of the code. However, how do we do `err`-checking then? Answer: enter the `MustXXXX` methods, which will panic if they encounter an error, and are intended for doing things like this:

```
var (
    semverRegex := regexp.MustCompile(SEMVER_REGEX)
)
```

(also you'd want `SEMVER_REGEX` to really be `semverRegexStr` so that it doesn't collide with the actual regex object)